lib/Models/Cesium.js

```diff
@@ -338,10 +339,35 @@ Cesium.prototype.zoomTo = function(viewOrExtent, flightDurationSeconds) {
     flightDurationSeconds = defaultValue(flightDurationSeconds, 3.0);
 
     if (viewOrExtent instanceof Rectangle) {
-        this.scene.camera.flyTo({
-            duration: flightDurationSeconds,
-            destination: viewOrExtent
+
+        var camera = this.scene.camera;
+
+        // Work out the destination that the camera would naturally fly to
+        var destinationCartesian = camera.getRectangleCameraCoordinates(viewOrExtent);
+        var destination = Ellipsoid.WGS84.cartesianToCartographic(destinationCartesian);
+
+        var terrainProvider = this.scene.globe.terrainProvider;
+        var level = 6; // A sufficiently coarse tile level that still has approximately accurate height
+        var positions = [Rectangle.center(viewOrExtent)];
+
+        // Perform an elevation query at the centre of the rectangle
+        sampleTerrain(terrainProvider, level, positions).then(function(results) {
```

Comment:
It would be good to return this promise, and then change `CatalogItem.zoomTo` to return the result of `terria.currentViewer.zoomTo`.  That way the `zoomTo` promise doesn't resolve until the zoom has actually started.